File: wallet-solution-mobile-app/onboarding/src/main/java/com/ekenya/rnd/dashboard/datadashboard/remote/APIService.java
package com.ekenya.rnd.dashboard.datadashboard.remote;


import com.ekenya.rnd.common.data.model.MainDataObject;
import com.ekenya.rnd.dashboard.datadashboard.model.LoginResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface APIService {
    @POST("MobileWebService")
    Call<LoginResponse> login(@Body MainDataObject user);


}

File: wallet-solution-mobile-app/onboarding/src/main/java/com/ekenya/rnd/onboarding/dataonboarding/api/ApiService.kt
package com.ekenya.rnd.onboarding.dataonboarding.api

import com.ekenya.rnd.common.data.model.MainDataObject
import com.ekenya.rnd.onboarding.dataonboarding.model.RegistrationResponse2
import com.ekenya.rnd.onboarding.dataonboarding.model.UserData
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST


interface  ApiService {
    @GET("users")
    suspend fun getUsers(): List<UserData>
    @POST("MobileWebService")
    suspend fun registerUser(@Body user: MainDataObject?): RegistrationResponse2
    @POST("MobileWebService")
    suspend fun ChangePassword(@Body user: MainDataObject?): Call<RegistrationResponse2?>?
    @POST("MobileWebService")
    suspend fun login(@Body user: MainDataObject?): Call<RegistrationResponse2?>?
    @POST("MobileWebService")
    suspend fun getWalletBalance(@Body user: MainDataObject?): Call<RegistrationResponse2?>?
}